extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -537,6 +540,13 @@ static <DATA_TYPE> void writeColumnSource(
                         maxValuesPerPage,
                         maxOriginalRowsPerPage,
                         pageCount);
+                if (!usedDictionary) {
+                    // Need to reset the state so that all values can again be read again for plain encoding
+                    if (case1) {
+                        rowStepGetter = originalRowsPerPage.iterator()::next;
+                        valuesStepGetter = rawItemCountPerPage.iterator()::next;
+                    }
+                }
```

Comment:
```suggestion
                if (!usedDictionary && case1) { /* Maybe rawItemCountPerPage != null instead of case1 */
                    // Need to reset the state so that all values can again be read again for plain encoding
                    rowStepGetter = originalRowsPerPage.iterator()::next;
                    valuesStepGetter = rawItemCountPerPage.iterator()::next;
                }
```